webrender/src/display_list_flattener.rs

```diff
@@ -582,33 +564,34 @@ impl<'a> DisplayListFlattener<'a> {
         &mut self,
         traversal: &mut BuiltDisplayListIter<'a>,
         pipeline_id: PipelineId,
-        item: &DisplayItemRef,
+        parent_spatial_node: SpatialNodeIndex,
+        origin: LayoutPoint,
         reference_frame: &ReferenceFrame,
-        clip_and_scroll_ids: &ClipAndScrollInfo,
         reference_frame_relative_offset: LayoutVector2D,
     ) {
         self.push_reference_frame(
             reference_frame.id,
-            Some(clip_and_scroll_ids.scroll_node_id),
-            clip_and_scroll_ids.clip_node_id,
+            Some(parent_spatial_node),
             pipeline_id,
             reference_frame.transform_style,
             reference_frame.transform,
             reference_frame.perspective,
-            reference_frame_relative_offset + item.rect().origin.to_vector(),
+            reference_frame_relative_offset + origin.to_vector(),
         );
 
         self.flatten_items(traversal, pipeline_id, LayoutVector2D::zero());
     }
 
+
     fn flatten_stacking_context(
         &mut self,
         traversal: &mut BuiltDisplayListIter<'a>,
         pipeline_id: PipelineId,
-        item: &DisplayItemRef,
         stacking_context: &StackingContext,
-        scroll_node_id: ClipId,
-        reference_frame_relative_offset: LayoutVector2D,
+        origin: LayoutPoint,
```

Comment:
Not really clear to me why you chose to replace the `item: &DisplayItemRef` with `origin: LayoutPoint, filters: ItemRange<FilterOp>` here but you didn't do the equivalent thing in e.g. `flatten_iframe`. Doesn't matter much functionally, but this is the sort of unrelated change that slows down my reviews...

Reply:
I figured that the internal methods shouldn't be working with `DisplayItemRef` and instead just receive the data they need, but ended up only fixing the spots I was working with. It was already not very consistent there, so I didn't make it worse. I think we should just follow-up with this later.